File: dataset/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Task36.java

import java.io.*;
import java.util.*;

public class Task36 {
    public static String readFileContents(String filename) {
        StringBuilder content = new StringBuilder();
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = reader.readLine()) != null) {
                content.append(line).append("\\n");
            }
        } catch (IOException e) {
            return "Error reading file: " + e.getMessage();
        }
        return content.toString();
    }
    
    public static void main(String[] args) {
        // Test case 1: Create and read a simple text file
        try {
            PrintWriter writer = new PrintWriter("test1.txt");
            writer.println("Hello, World!");
            writer.println("This is test file 1.");
            writer.close();
            System.out.println("Test Case 1:");
            System.out.println(readFileContents("test1.txt"));
        } catch (Exception e) {
            System.out.println("Error in test case 1: " + e.getMessage());
        }
        
        // Test case 2: Create and read a file with multiple lines
        try {
            PrintWriter writer = new PrintWriter("test2.txt");
            writer.println("Line 1");
            writer.println("Line 2");
            writer.println("Line 3");
            writer.close();
            System.out.println("Test Case 2:");
            System.out.println(readFileContents("test2.txt"));
        } catch (Exception e) {
            System.out.println("Error in test case 2: " + e.getMessage());
        }
        
        // Test case 3: Empty file
        try {
            PrintWriter writer = new PrintWriter("test3.txt");
            writer.close();
            System.out.println("Test Case 3 (Empty file):");
            System.out.println(readFileContents("test3.txt"));
        } catch (Exception e) {
            System.out.println("Error in test case 3: " + e.getMessage());
        }
        
        // Test case 4: File with special characters
        try {
            PrintWriter writer = new PrintWriter("test4.txt");
            writer.println("Special chars: @#$%^&*()");
            writer.println("Numbers: 123456789");
            writer.close();
            System.out.println("Test Case 4:");
            System.out.println(readFileContents("test4.txt"));
        } catch (Exception e) {
            System.out.println("Error in test case 4: " + e.getMessage());
        }
        
        // Test case 5: Non-existent file
        System.out.println("Test Case 5 (Non-existent file):");
        System.out.println(readFileContents("nonexistent.txt"));
    }
}

File: dataset/Vanilla - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task36.cpp

#include <iostream>
#include <fstream>
#include <sstream>
#include <string>

std::string readFileContents(const std::string& filename) {
    std::ifstream file(filename);
    if (!file.is_open()) {
        return "Error reading file: Unable to open file";
    }
    
    std::stringstream buffer;
    buffer << file.rdbuf();
    file.close();
    
    return buffer.str();
}

int main() {
    // Test case 1: Create and read a simple text file
    {
        std::ofstream outFile("test1.txt");
        outFile << "Hello, World!" << std::endl;
        outFile << "This is test file 1." << std::endl;
        outFile.close();
        std::cout << "Test Case 1:" << std::endl;
        std::cout << readFileContents("test1.txt") << std::endl;
    }
    
    // Test case 2: Create and read a file with multiple lines
    {
        std::ofstream outFile("test2.txt");
        outFile << "Line 1" << std::endl;
        outFile << "Line 2" << std::endl;
        outFile << "Line 3" << std::endl;
        outFile.close();
        std::cout << "Test Case 2:" << std::endl;
        std::cout << readFileContents("test2.txt") << std::endl;
    }
    
    // Test case 3: Empty file
    {
        std::ofstream outFile("test3.txt");
        outFile.close();
        std::cout << "Test Case 3 (Empty file):" << std::endl;
        std::cout << readFileContents("test3.txt") << std::endl;
    }
    
    // Test case 4: File with special characters
    {
        std::ofstream outFile("test4.txt");
        outFile << "Special chars: @#$%^&*()" << std::endl;
        outFile << "Numbers: 123456789" << std::endl;
        outFile.close();
        std::cout << "Test Case 4:" << std::endl;
        std::cout << readFileContents("test4.txt") << std::endl;
    }
    
    // Test case 5: Non-existent file
    std::cout << "Test Case 5 (Non-existent file):" << std::endl;
    std::cout << readFileContents("nonexistent.txt") << std::endl;
    
    return 0;
}
